scanner/datastore/postgres/external_index_store.go

```diff
@@ -0,0 +1,62 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/jackc/pgx/v5/stdlib"
+	"github.com/pkg/errors"
+	"github.com/quay/claircore"
+	"github.com/remind101/migrate"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/scanner/datastore/postgres/migrations"
+)
+
+type ExternalIndexStore interface {
+	StoreIndexReport(
+		ctx context.Context,
+		hashID string,
+		scannerVersion string,
+		indexReport *claircore.IndexReport,
+		expiration time.Time,
+		versionCmpFn func(iv string) bool,
```

Comment:
super nit: consider carrying the name `shouldUpdateExternalIndexReport` (or similar, such as `shouldUpdateStoredReportFn`) forward as it is more descriptive then `versionCmpFn`.